### PR TITLE
ESLint upgrade from JSHint 

### PR DESCRIPTION
**Pull request by Alkarex**
* commit corresponding `package.json`
* `npm run fix` for automatic JS and CSS fixes (this is how all the changes have been made in this PR)

Documentation:
* https://eslint.org
* Using most of the rules from https://github.com/standard/standardx , based on https://standardjs.com/rules.html

### CODE REVIEW

**Comment by Frenzie** [comment]
Are you sure ESLint checks for all the same things? I'd be inclined to say they're complementary.

**Comment by Alkarex** [comment]
@Frenzie I am not aware of anything in JSHint not available in ESLint. but let me know if you are worried about any specific point

File: .stylelintrc [hunk 1]
```diff
@@ -33,11 +33,9 @@
     "no-eol-whitespace": true,
     "property-no-vendor-prefix": true,
     "rule-empty-line-before": [
-      "always",
-      "except": [
```
Comment (Alkarex) [inline .stylelintrc:37]: There was a typo in this JSON file, making it invalid (found when updating the NPM commands)

**Comment by Frenzie** [comment]
I seem to recall ESLint not warning about missing break/return statements, variables referenced from a higher scope that could lead to confusing semantics, and various other things. But that was a few years ago.

**Comment by Frenzie** [comment]
Of course JSHint comes with various annoying defaults, like its ridiculous "use dot notation" nonsense. But they've actually recognized since that they should just focus on code quality and get rid of aesthetic preferences (definitely ESLint for that).

**Comment by Alkarex** [comment]
Regarding `break`, if I remove one:
image

If there is any rule that you believe is either lacking or annoying, let's have a look. From what I can see, ESLint is now a large superset of JSHint.

**Comment by Frenzie** [comment]
Maybe just don't delete the config yet so we can easily run them side by side for a while?

**Comment by Alkarex** [comment]
> Maybe just don't delete the config yet so we can easily run them side by side for a while?

@Frenzie Ok, I have restored JSHint for now and made sure it can run fine as well.

**Comment by Frenzie** [comment]
Cool, thanks :+1:

**Comment by Alkarex** [comment]
Any comment from @aledeg ?

**Comment by aledeg** [comment]
Sorry for the lack of activity. I have so many personal project going on at the same time.
I do not have comments on that matter.

**Comment by Alkarex** [comment]
No worries @aledeg :-)

**Pull request merged**